### PR TITLE
homepage module

### PR DESCRIPTION
**Pull request by RashiTiwari21**

### CODE REVIEW

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	 *
+	 * @param expectedHeading The expected heading of the Related Content section.
+	 * @param index The index of the content card to click on.
+	 * @param expectedUrl The expected URL after clicking on the content card.
+	 */
+	public void verifyRelatedContent(String expectedHeading, String index, String expectedUrl) {
+		extentInfoLog("Related Content Section is displayed : ",Action.isDisplayed(driver,this.relatedContent));
+		String actualHeading = this.relatedContent.getText();
+		Assert.assertEquals(actualHeading, expectedHeading);
+		System.out.println(this.getRelatedContentCard(index));
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: @RashiTiwari21 Please remove system log. If required use the extentInfo log instead

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		Action.performActionwithExtentInfoLog(this.getRelatedContentCard(index), "click",
+				"Clicking on : " + getRelatedContentCard(index).getText());
+		Action.verifyPageUrl(expectedUrl);
+		extentInfoLog("URL Verified!","");
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: @RashiTiwari21 Append the url verified in the log
extentInfoLog("URL Verified : ",expectedUrl);

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	 * @param expectedUrl The expected URL after clicking on the "Learn More" button.
+	 */
+	public void verifyAllergySection(String sectionName, String expectedHeading,String sectionBtn, String expectedUrl) {
+		extentInfoLog("Allergy Section is displayed : ",Action.isDisplayed(driver,this.getAllergiesSection(sectionName)));
+		String actualHeading = this.getAllergiesSection(sectionName).getText();
+		Assert.assertEquals(actualHeading, expectedHeading);
+		Action.performActionwithExtentInfoLog(this.getAllergiesLearnMore(sectionBtn), "click",
+				"Clicking on : " + getAllergiesLearnMore(sectionBtn).getText());
+		Action.verifyPageUrl(expectedUrl);
+		extentInfoLog("Successfully verified URL!","");
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: @RashiTiwari21 Append the url verified in the log
extentInfoLog("Successfully verified URL : ",expectedUrl);

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	 * @param expURL The expected URL after clicking on the specified product.
+	 */
+	public void moreOnSection(String expHeading,String pdctName, String	expURL) {
+		extentInfoLog("More on Benadryl Section is displayed : ",Action.isDisplayed(driver,this.moreOnSectn));
+		String actualHeading = this.moreOnSectn.getText();
+		Assert.assertEquals(actualHeading, expHeading);
+		Action.performActionwithExtentInfoLog(this.moreOnSectn(pdctName), "click",
+				"Clicking on : " + moreOnSectn(pdctName).getText());
+		Action.verifyPageUrl(expURL);
+		extentInfoLog("Successfully verified URL!","");
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: @RashiTiwari21 Append the url verified in the log
extentInfoLog("Successfully verified URL : ",expectedUrl);

File: BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	public void HOME_verifyMoreOnSection(String testcase, String execution, String expHeading,String pdctName, String expURL) {
+		test = test.createNode(testcase);
+		if (execution.equalsIgnoreCase(defaultFlag)) {
+			home.moreOnSection(expHeading, pdctName,expURL);
+		} else {
+			throw new SkipException("Test skipped : " + testcase);
+		}
+	}
+
+
```
Comment (goutamnaik007) [inline BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java]: @RashiTiwari21 remove the extra blank line and format the source code

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 1 above)
Comment (RashiTiwari21) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: fixed

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 2 above)
Comment (RashiTiwari21) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: fixed

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 3 above)
Comment (RashiTiwari21) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: fixed

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 4 above)
Comment (RashiTiwari21) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: fixed

File: BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java (hunk 5 above)
Comment (RashiTiwari21) [reply BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java]: fixed

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 1 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: Verified @RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 2 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: Verified @RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 3 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: Verified @RashiTiwari21

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java (hunk 4 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/HomePage.java]: Verified @RashiTiwari21

File: BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java (hunk 5 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/test/java/com/jnj/testcases/HomePageTest.java]: Verified @RashiTiwari21

**Pull request merged**